app/controllers/api/offers_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class Api::OffersController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    offer = Offer.create(check_and_lump_parameters)
+    if offer.persisted?
+      render json: { message: 'Your offer has been sent!' }
+    else
+      render json: { message: offer.errors.messages.to_a.flatten.join(' ').capitalize }, status: 422
```

Comment:
cleaner with .full_message?!

Reply:
Done, although had to do full_messages and extract from array, as full_message method didn't work.